kafka/src/kafka/impl/consumer_impl.cpp

```diff
@@ -292,6 +293,59 @@ void ConsumerImpl::Commit() { rd_kafka_commit(consumer_.GetHandle(), nullptr, /*
 
 void ConsumerImpl::AsyncCommit() { rd_kafka_commit(consumer_.GetHandle(), nullptr, /*async=*/1); }
 
+OffsetRange ConsumerImpl::GetOffsetRange(const std::string& topic, std::int32_t partition) const {
+    OffsetRange offset_range;
+    auto err = rd_kafka_query_watermark_offsets(
+        consumer_.GetHandle(),
+        topic.с_str(),
+        partition,
+        &offset_range.low,
+        &offset_range.high,
+        /*timeout_ms=*/-1
```

Comment:
add `std::optional<std::milliseconds>` argument to allow user to provide a timeout for the call